ReviveOverlay/oculusoauthtokencontroller.cpp

```diff
@@ -0,0 +1,102 @@
+#include "oculusoauthtokencontroller.h"
+#include "trayiconcontroller.h"
+#include "windowsservices.h"
+#include <stdlib.h>
+
+#include <QCoreApplication>
+
+COculusOauthTokenController *COculusOauthTokenController::SharedInstance()
+{
+    static COculusOauthTokenController *instance = nullptr;
+	if ( !instance )
+	{
+        instance = new COculusOauthTokenController();
+	}
+	return instance;
+}
+
+COculusOauthTokenController::COculusOauthTokenController()
+    : m_strAccessToken()
+    , m_oculusOfflineDBPath(qgetenv("APPDATA") + "\\Oculus\\sessions\\_oaf\\")
```

Comment:
The standard Qt way to retrieve the AppData location is to use `QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)`.

Also, it's recommended to use forward slashes for Qt as it will automatically convert those to backslashes if needed.